client.go

```diff
@@ -15,12 +21,30 @@ const (
 )
 
 type Client struct {
-	cfg *Config
+	cfg    *Config
+	logger *logrus.Logger
 }
```

Comment:
We do not need instance/reference of logger into Client struct. We need only a flag e.g., debuglevel, which default value is` False`. The user of the client-go lib can enable to show more debugging messages.